src/Providers/AuthServiceProvider.php

```diff
@@ -84,7 +85,13 @@ public function boot()
         });
 
         Gate::before(function ($user, $ability) {
-            return optional(User::fromUser($user))->isSuper() ? true : null;
+            Permission::boot();
```

Comment:
While I think this approach is broadly on the right track, I don't really want to be booting permissions here... 

Although, I'm not sure what the best alternative is... 🤔
* Without booting permissions here, users can't access the CP because the `Authorize` middleware checks if the user has the relevant permissions, however it happens _before_ the `BootPermissions` middleware is called.
* We can't reorder the middleware, otherwise it might cause issues with translations in permission labels, which #7343 solved.

Reply:
What's wrong with booting in there? It only happens when the closure is executed.

Reply:
I guess I was just thinking it was weird since we're already booting them in a middleware, and that trying to boot them multiple times would hurt performance?

Maybe I'm wrong though, and it's fine?

Reply:
We could make `Permissions::boot()` return early and do nothing if it's already been booted once.